tests/migration/test_config_map_data_handler.py

```diff
@@ -0,0 +1,744 @@
+            # call method under test
+            cmdh = ConfigMapDataHandler()
+            cmdh.create_product_config_maps(PROD_CM_DATA_LIST_EXPECTED)
+
+            dummy_prod_cm_names = ['cray-product-catalog-hfp-firmware', 'cray-product-catalog-analytics']
+
+            self.mock_k8api_create.assert_has_calls(calls=[  # Create ConfigMap called twice
+                call(
+                    dummy_prod_cm_names[0], PRODUCT_CATALOG_CONFIG_MAP_NAMESPACE,
+                    PROD_CM_DATA_LIST_EXPECTED[0], PRODUCT_CATALOG_CONFIG_MAP_LABEL), call().__bool__(),
+                call(
+                    dummy_prod_cm_names[1], PRODUCT_CATALOG_CONFIG_MAP_NAMESPACE,
+                    PROD_CM_DATA_LIST_EXPECTED[1], PRODUCT_CATALOG_CONFIG_MAP_LABEL), call().__bool__(),
+            ]
+            )
+
+            # Verify the exact log message
+            self.assertEqual(
+                captured.records[0].getMessage(),
+                f"Created product ConfigMap {PRODUCT_CATALOG_CONFIG_MAP_NAMESPACE}/{dummy_prod_cm_names[0]}")
+
+            self.assertEqual(
+                captured.records[1].getMessage(),
+                f"Created product ConfigMap {PRODUCT_CATALOG_CONFIG_MAP_NAMESPACE}/{dummy_prod_cm_names[1]}")
+
+    def test_create_second_product_config_map_failed(self):
+        """ Validating failed to create product config maps """
+
+        # mock some additional functions
+        self.mock_v1_object_Meta_mig = patch('cray_product_catalog.migration.kube_apis.V1ObjectMeta').start()
+
+        with self.assertLogs() as captured:
+            self.mock_k8api_create.side_effect = [True, False]
+
+            # call method under test
+            cmdh = ConfigMapDataHandler()
+            cmdh.create_product_config_maps(PROD_CM_DATA_LIST_EXPECTED)
+
+            dummy_prod_cm_names = ['cray-product-catalog-hfp-firmware', 'cray-product-catalog-analytics']
+
+            self.mock_k8api_create.assert_has_calls(calls=[  # Create ConfigMap called twice
+                call(
+                    dummy_prod_cm_names[0], PRODUCT_CATALOG_CONFIG_MAP_NAMESPACE,
+                    PROD_CM_DATA_LIST_EXPECTED[0], PRODUCT_CATALOG_CONFIG_MAP_LABEL),
+                call(
+                    dummy_prod_cm_names[1], PRODUCT_CATALOG_CONFIG_MAP_NAMESPACE,
+                    PROD_CM_DATA_LIST_EXPECTED[1], PRODUCT_CATALOG_CONFIG_MAP_LABEL),
+            ]
+            )
+
+            # Verify the exact log message
+            self.assertEqual(
+                captured.records[0].getMessage(),
+                f"Created product ConfigMap {PRODUCT_CATALOG_CONFIG_MAP_NAMESPACE}/{dummy_prod_cm_names[0]}")
+
+            self.assertEqual(
+                captured.records[1].getMessage(),
+                f"Failed to create product ConfigMap {PRODUCT_CATALOG_CONFIG_MAP_NAMESPACE}/{dummy_prod_cm_names[1]}")
+
+    def test_create_first_product_config_map_failed(self):
+        """ Validating failed to create product config maps """

```

Comment:
change the doc string to address first configmap

Reply:
changed